### PR TITLE
Mount capacitor js to web view onLoadResource

### LINKED ISSUE #4373: Can't use barcode scanner for Outbound Shipments or update item details (Android)
## What went wrong? 😲

While using V2.1.RC6 .apk,  @louisaw123 and I noticed the barcode scanning buttons are not available at:

- Outbound shipments: 'scan' button to add item lines
- Stock > View Stock: 'scan' icon to add batch and expiry date are not working

It is available on V2.1, which prob leads us to think it disappeared somewhere in between these versions.

- Open mSupply Version: V2.1.RC6
- Platform: Android

### PR DESCRIPTION
**Pull request by andreievg**
Fixes #4373

# 👩🏻‍💻 What does this PR do?

I found that native capacitor plugins were not always available, sometimes they were there and sometimes not. On tablet with dev options on and usb debugging plugged in I could inspect app while it was running with chrome using 'chrome://inspect'. Managed to see a pattern, when adding [console.log here](https://github.com/msupply-foundation/open-msupply/blob/f8dacaf0164d5e8c3442ff54067870ae40ecf19d/client/packages/android/app/src/main/java/org/openmsupply/client/ExtendedWebViewClient.java#L66), and it seem when the log was showing after bugsnag was mounted, web app was not able to recognise native functionality, but when the log was showing before bugsnag mounted log, native functionality worked as expected. This lead me to thing that capacitor JS injection was happening late, which was confirmed by the fix.

No 100% happy about it all, can discuss future improvement on this front a bit later, but I think this change is safe (relative to what is currently happening

## 💌 Any notes for the reviewer?

We had to move to using evaluateJavascript because existing capacitor 'injection', that happens here:

https://github.com/ionic-team/capacitor/blob/f4e7f19f186e334404b2cd0decc3205e57bf4469/android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java#L372-L381

Relies on below method which fails on self signed certs:

https://github.com/ionic-team/capacitor/blob/f4e7f19f186e334404b2cd0decc3205e57bf4469/android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java#L348

Some extra material: https://github.com/ionic-team/capacitor/discussions/6166

# 🧪 Testing

On tablet open stock view for one batch, next to barcode you should see a scan icon. This icon is sometimes missing (if you restart and start app multiple times you should see it, for me it happens quite oftern)

### CODE REVIEW

**Comment by andreievg** [comment]
I think fix would solve this issue: https://github.com/msupply-foundation/open-msupply/issues/4069

File: client/packages/android/app/src/main/java/org/openmsupply/client/ExtendedWebViewClient.java [hunk 1]
```diff
@@ -71,9 +77,20 @@ public void onPageStarted(WebView webView,
                     cordovaPluginsJS +
                     "\n\n";
 
-            // From JSInjector.getScriptString()
             // .post to run on UI thread
-            webView.post(() -> webView.evaluateJavascript(fullScript, null));
+            webView.post(() -> {
+                // To only run once on a page we check for existence ot nativeBridge
```
Comment (jmbrunskill) [inline client/packages/android/app/src/main/java/org/openmsupply/client/ExtendedWebViewClient.java:82]:
```suggestion
                // To only run once on a page we check for existence of nativeBridge
```

File: client/packages/android/app/src/main/java/org/openmsupply/client/ExtendedWebViewClient.java [hunk 2]
```diff
@@ -26,10 +27,14 @@ public ExtendedWebViewClient(Bridge bridge) {
     // but since it manually uses net.URL to fetch the content of request, this
     // fails for self signed certificates
     // and plugin definitions etc is not injected
+
+    // Using onLoadResource vs onPageStarted because sometimes onPageStarted is too late
+    // but with onLoadResource we need to make sure that injection happens only one in a web view
```
Comment (jmbrunskill) [inline client/packages/android/app/src/main/java/org/openmsupply/client/ExtendedWebViewClient.java:32]:
```suggestion
    // but with onLoadResource we need to make sure that injection happens only once in a web view
```

**Review by jmbrunskill — APPROVED** [review]
Looks legit to me, haven't got any experience in this area though.
Nice find!

**Comment by andreievg** [comment]
Thanks @jmbrunskill for review !

**Pull request merged**